crates/ty_python_semantic/resources/mdtest/pep695_type_aliases.md

```diff
@@ -244,6 +244,16 @@ def f(x: IntOr, y: OrInt):
         reveal_type(x)  # revealed: Never
     if not isinstance(y, int):
         reveal_type(y)  # revealed: Never
+
+# TODO emit a diagnostic on this line
+type Itself = Itself
```

Comment:
Does this fix also work for compound types like list[Itself]?

Reply:
Another case that may be worth testing is multi-step direct recursion:

```
type A = B
type B = A
```